### PR TITLE
[tests] Add support for CustomAdminOptions; pass along InstrumentOptions on server create

### PR DESCRIPTION
**Pull request by nbroyles**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/dbnode/integration/setup.go [hunk 1]
```diff
@@ -1035,6 +1035,13 @@ func newClients(
 		verificationAdminOpts = verificationAdminOpts.SetEncodingProto(prototest.ProtoPools.EncodingOpt).(client.AdminOptions)
 	}
 
+	if len(opts.CustomAdminOptions()) > 0 {
+		for _, opt := range opts.CustomAdminOptions() {
```
Comment (wesleyk) [inline src/dbnode/integration/setup.go:1039]: maybe optFn? Found myself scratching my head for a sec on this. Feel free to ignore if this is the standard

**Review by wesleyk — APPROVED** [review]

File: src/dbnode/integration/setup.go [hunk 2]
```diff
@@ -1035,6 +1035,13 @@ func newClients(
 		verificationAdminOpts = verificationAdminOpts.SetEncodingProto(prototest.ProtoPools.EncodingOpt).(client.AdminOptions)
 	}
 
+	if len(opts.CustomAdminOptions()) > 0 {
```
Comment (ryanhall07) [inline src/dbnode/integration/setup.go:1038]: you don't need the additional if

**Review by ryanhall07 — APPROVED** [review]

File: src/dbnode/integration/setup.go (hunk 2 above)
Comment (nbroyles) [reply src/dbnode/integration/setup.go:1038]: lol, smh 🤦

File: src/dbnode/integration/setup.go (hunk 1 above)
Comment (nbroyles) [reply src/dbnode/integration/setup.go:1039]: It _is_ the standard but that doesn't mean we should keep doing it :). Will update.

**Pull request merged**